libkbfs/folder_update_prepper.go

```diff
@@ -1143,11 +1153,15 @@ func (fup folderUpdatePrepper) prepUpdateForPaths(ctx context.Context,
 			// Updates go in the first one.
 			for _, update := range unmergedResOp.allUpdates() {
 				chain, isMostRecent := unmergedChains.byMostRecent[update.Ref]
+				isDeleted := false
 				alreadyUpdated := false
 				if isMostRecent {
+					isDeleted = unmergedChains.isDeleted(chain.original) ||
+						unmergedChains.toUnrefPointers[update.Ref]
 					_, alreadyUpdated = updates[chain.original]
 				}
-				if newBlocks[update.Ref] || (isMostRecent && !alreadyUpdated) {
+				if newBlocks[update.Ref] ||
+					(isMostRecent && !isDeleted && !alreadyUpdated) {
```

Comment:
If an update in a previous `resolutionOp` points to a block that's being deleted in this batch, we need to avoid copying that update over into the new `resolutionOp`.